nuxeo-tree-snapshot-core/src/main/java/org/nuxeo/snapshot/SnapshotableAdapter.java

```diff
@@ -132,19 +137,38 @@ public Snapshot createSnapshot(VersioningOption option) {
 
         DocumentModelList children = doc.getCoreSession().getChildren(doc.getRef());
```

Comment:
This should be removed if the goal is to replace this call with something more efficient.